nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/services/azure/storage/TestADLSSecureCredentialsControllerService.java

```diff
@@ -0,0 +1,238 @@
+
+        runner.setProperty(credentialsService, AzureStorageUtils.ADLS_ENDPOINT_SUFFIX, "");
+        runner.assertNotValid(credentialsService);
+    }
+
+    @Test
+    public void testNotValidWithWhitespaceEndpointSuffix() {
+        configureAccountName();
+        configureAccountKey();
+        runner.setProperty(credentialsService, AzureStorageUtils.ADLS_ENDPOINT_SUFFIX, " ");
+
+        runner.assertNotValid(credentialsService);
+    }
+
+    @Test
+    public void testValidWithAccountNameAndAccountKey() throws InitializationException {
+        configureAccountName();
+        configureAccountKey();
+        configureKeyVaultConnectionService();
+
+        runner.assertValid(credentialsService);
+    }
+
+    @Test
+    public void testValidWithAccountNameAndSasToken() throws InitializationException {
+        configureAccountName();
+        configureSasToken();
+        configureKeyVaultConnectionService();
+
+        runner.assertValid(credentialsService);
+    }
+
+    @Test
+    public void testGetCredentialsDetailsWithAccountKey() throws Exception {
+        // GIVEN
+        configureAccountName();
+        configureAccountKey();
+        configureKeyVaultConnectionService();
+
+        runner.enableControllerService(credentialsService);
+
+        // WHEN
+        ADLSCredentialsDetails actual = credentialsService.getCredentialsDetails(new HashMap<>());
+
+        // THEN
+        assertEquals(ACCOUNT_NAME_SECRET_VALUE, actual.getAccountName());
+        assertEquals(ACCOUNT_KEY_SECRET_VALUE, actual.getAccountKey());
+        assertNull(actual.getSasToken());
+        assertFalse(actual.getUseManagedIdentity());
+        assertNotNull(actual.getEndpointSuffix());
+    }
+
+    @Test
+    public void testGetCredentialsDetailsWithSasToken() throws Exception {
+        // GIVEN
+        configureAccountName();
+        configureSasToken();
+        configureKeyVaultConnectionService();
+
+        runner.enableControllerService(credentialsService);
+
+        // WHEN
+        ADLSCredentialsDetails actual = credentialsService.getCredentialsDetails(new HashMap<>());
+
+        // THEN
+        assertEquals(ACCOUNT_NAME_SECRET_VALUE, actual.getAccountName());
+        assertEquals(SAS_TOKEN_SECRET_VALUE, actual.getSasToken());
+        assertNull(actual.getAccountKey());
+        assertFalse(actual.getUseManagedIdentity());
+        assertNotNull(actual.getEndpointSuffix());
+    }
+
+    @Test
+    public void testGetCredentialsDetailsWithSetEndpointSuffix() throws Exception {
+        // GIVEN
+        configureAccountName();
+        configureAccountKey();
+        configureEndpointSuffix();
+        configureKeyVaultConnectionService();
+
+        runner.enableControllerService(credentialsService);
+
+        // WHEN
+        ADLSCredentialsDetails actual = credentialsService.getCredentialsDetails(new HashMap<>());
+
+        // THEN
+        assertEquals(END_POINT_SUFFIX_SECRET, actual.getEndpointSuffix());
+    }
+
```

Comment:
What happens when the client service returns an empty or blank string? I think we should add test cases for those.